#!/usr/bin/env python
# FSD configuration

PACKAGE = 'pandora_vision_obstacle'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0,    "An In.....", 50, 0,  100)
#                    name        type   level  description  def min max

#------------------------------- Show Images -----------------------------
gen.add("show_respective_barrel", bool_t, 0,"", False)
gen.add("show_valid_barrel", bool_t, 0,"", False)
#
##------------------------ Barrel FSD Thresholds ----------------------------
gen.add("fsd_canny_thresh_1", int_t, 0,"", 0, 0, 200)
gen.add("fsd_canny_thresh_2", int_t, 0,"", 27, 0, 200)
gen.add("fsd_min_pair_dist", int_t, 0,"", 100, 0, 500)
gen.add("fsd_max_pair_dist", int_t, 0,"", 640, 0, 640)
gen.add("fsd_no_of_peaks", int_t, 0,"", 1, 0, 10)
gen.add("roi_variance_thresh", double_t, 0,"", 60.0, 0.0, 255.0)
gen.add("differential_depth_unsymmetry_thresh", double_t, 0,"", 30.0, 0.0, 255.0)
gen.add("symmetry_line_depth_difference_thresh", double_t, 0,"", 11.0, 0.0, 20.0)
gen.add("curve_approximation_max_epsilon", double_t, 0,"", 1600.0, 0.0, 200000.0)
gen.add("min_circle_overlapping", double_t, 0,"", 0.25, 0.0, 1.0)
gen.add("max_corner_thresh", double_t, 0,"", 35.0, 0.0, 255.0)

exit(gen.generate(PACKAGE, "barrel_node", "barrel_node"))
